experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+    scan boundary if any. A time offset between the pulses
+    starting may be set (seq_timer in exp_slice). exp_slice A
+    and B will have integrations that run at the same time.
+"""
 
 
+debug_flag = True  # TODO move this somewhere central. config? - then all files could use it
+
 class ExperimentPrototype(object):
-    """Class combining control program objects, defining how they
-    interface and some overall metadata
+    """A prototype experiment class composed of metadata, including experiment slices (exp_slice) 
+    which are dictionaries of radar parameters. Basic, traditional experiments will be composed 
+    of a single slice. More complicated experiments will be composed of multiple slices that 
+    interface in one of four pre-determined ways, as described in more detail below. 
+    
+    This class is used via inheritance to create experiments.
+    
+    Some variables shouldn't be changed after init, and their properties do not have setters. 
+    Some variables in this class are given property setters and can be modified with a class 
+    method entitled 'update' in your experiment class. 
 
-    :param cpid: unique id (RCP number)
-    :param cponum: number of CPObjects in this control program.
+    :param cpid: unique id necessary for each control program (experiment)
     :
-    """ # REVIEW #26 - the ordering of arguments in the documentation isn't consistent with the ordering in the __init__ function. cpid should go first based on the comment
+    """
+
+    __slice_keys = ["slice_id", "cpid", "txantennas", "rx_main_antennas",
+                    "rx_int_antennas", "pulse_sequence", "pulse_shift", "mpinc",
+                    "pulse_len", "nrang", "frang", "intt", "intn", "beam_angle",
+                    "beam_order", "scanboundflag", "scanbound", "txfreq", "rxfreq",
+                    "clrfrqrange", "acf", "xcf", "acfint", "wavetype", "seqtimer"]
+
+    __hidden_slice_keys = ['rxonly', 'clrfrqflag']
+
+    """ The slice keys are described as follows: 
+    
+    DESCRIPTION OF SLICE KEYS
+
+slice_id : The ID of this object. An experiment can have multiple objects.
+cpid: The ID of the experiment, consistent with existing radar control programs.
+txantennas: The antennas to transmit on, default is all antennas given max number from config.
+rx_main_antennas: The antennas to receive on in main array, default = all antennas given max number from config.
+rx_int_antennas : The antennas to receive on in interferometer array, default is all antennas given max number from config.
+pulse_sequence: The pulse sequence timing, given in quantities of mpinc, for example normalscan = [0, 14, 22, 24, 27, 31, 42, 43]
+mpinc: multi-pulse increment in us, Defines minimum space between pulses.
+pulse_shift: Allows phase shifting between pulses. Built in for a capability Ashton would like to use within a pulse sequence. Default all zeros for all pulses in pulse_sequence.
```

Comment:
Probably don't need to specify this is for Ashton? I think I told Pasha about this and he said its a good idea to use as well.